src/main/java/io/connect/scylladb/ScyllaDbSinkConnector.java

```diff
@@ -104,4 +107,71 @@ public ConfigDef config() {
   public String version() {
     return VersionUtil.getVersion();
   }
+
+  @Override
+  public Config validate(Map<String, String> connectorConfigs){
+    // Do the usual validation, field by field.
+    Config result = super.validate(connectorConfigs);
+
+    ConfigValue userConfig = getConfigValue(result, ScyllaDbSinkConnectorConfig.USERNAME_CONFIG);
+    ConfigValue passwordConfig = getConfigValue(result, ScyllaDbSinkConnectorConfig.PASSWORD_CONFIG);
+    if (userConfig.value() == null && passwordConfig.value() != null) {
+      userConfig.addErrorMessage("Username not provided, even though password is.");
+    }
+    if((userConfig.value() != null) && (passwordConfig.value() == null)){
+      userConfig.addErrorMessage("Password not provided, even though username is.");
+    }
+
+    ConfigValue securityEnable = getConfigValue(result, ScyllaDbSinkConnectorConfig.SECURITY_ENABLE_CONFIG);
+    if(securityEnable.value().equals(true)){
+      if(userConfig.value() == null){
+        userConfig.addErrorMessage("Username is required with security enabled.");
+      }
+      if(passwordConfig.value() == null){
+        passwordConfig.addErrorMessage("Password is required with security enabled.");
+      }
+    }
+
+    boolean hasErrors = result.configValues().stream().anyMatch(c -> !(c.errorMessages().isEmpty()) );
+    if(!hasErrors)
+    {
+      // Assuming that this config gets parsed and validated the same way as `Config result`
+      config = new ScyllaDbSinkConnectorConfig(connectorConfigs);
+      // Make trial connection
+      ScyllaDbSessionFactory sessionFactory = new ScyllaDbSessionFactory();
+      try (ScyllaDbSession session = sessionFactory.newSession(config)) {
```

Comment:
It seems that `newSession` only creates a new session, without actually connecting. Therefore if you configure a wrong (but valid) address, etc: "123.123.123.123" it doesn't return any validation error.

Reply:
I've changed ScyllaDbSessioonFactory so that it calls .connect now and if host is unreachable then this should be the result now:
![validate](https://user-images.githubusercontent.com/36934780/155000557-21ff4de3-87f0-44f1-bab1-4f4570340a73.png)